sympy/core/add.py

```diff
@@ -576,6 +580,8 @@ def _eval_is_positive(self):
             v = _monotonic_sign(a)
             if v is not None:
                 s = v + c
+                if s == self:
+                    return None
                 if s.is_positive and a.is_nonnegative:
```

Comment:
You can see in the other assumptions, I had to add checks if `s == self`, or else it recurses trivially (the cache prevents it from being infinite). But here the check also does something with `a`, and I can't figure out what the correct logic is. I believe if you remove the above `s == self` block and try to import with the cache off (`SYMPY_USE_CACHE=no`), you will see the issue.